main/static/script.js

```diff
@@ -16,20 +19,47 @@ module.controller("mainCtrl", function($scope, $http) {
 		    $scope.queryResult = data;
 		  });
 	};
+
 	$scope.$watch("queryResult", resultUpdate);
+
+	function readHash() {
+		var urlQuery = window.location.hash
+		if (urlQuery != "") {
+			// Drop the leading '#'
+			urlQuery = urlQuery.substring(1);
+			// The remainder is the query to perform.
+			// Decode it (if neccesarry).
+			$scope.queryText = decodeURIComponent(urlQuery);
+			$scope.onSubmitQuery();
+		}
+	}
+
+	readHash(); // Read the current hash (if present).
+
+	// Whenever the user changes the hash (such as pasting in a new URL), call readHash() again.
+	window.addEventListener("hashchange", readHash, false);
 });
 
 var chart;
 
+
 function onload() {
 	google.load('visualization', '1.0', {'packages':['corechart']});
+	google.setOnLoadCallback(function(){ chartsReady = true; });
 }
 
 function dateFromIndex(index, timerange) {
 	return new Date(timerange.start + timerange.resolution * index);
 }
 
+
+
 function resultUpdate(object) {
+	if (!chartsReady) {
+		// Ask again in 30ms.
+		setTimeout(function(){resultUpdate(object);}, 10);
```

Comment:
we should probably queue this up in some array, and call this on `setOnLoadCallback`.